pointers/unique_ptr.h

```diff
@@ -0,0 +1,110 @@
+#include <cstddef>
+#include <utility>
+
+template<typename T, typename Deleter = void(*)(T*)>
+class unique_ptr {
```

Comment:
please enclose your unique_ptr in a namespace to avoid having collision with std's unique_ptr (if someone would write `using namespace std`, compiler won't know which unique_ptr to use